packages/amplify-e2e-core/src/utils/rds.ts

```diff
@@ -449,6 +449,115 @@ export const setupRDSClusterAndData = async (config: RDSConfig, queries?: string
   };
 };
 
+export const clearRDSClusterData = async (clusterInfo: ClusterInfo, region: string): Promise<void> => {
+  const client = new RDSDataClient({ region });
+
+  // Get all table names
+  const tableQuery = `
+    SELECT table_name 
+    FROM information_schema.tables 
+    WHERE table_schema = 'public'
+      AND table_type = 'BASE TABLE'
+      AND table_catalog = '${clusterInfo.dbName}'
+      AND table_name LIKE 'e2e_test_%';
+  `;
+  const tableQueryInput: ExecuteStatementCommandInput = {
+    resourceArn: clusterInfo.clusterArn,
+    secretArn: clusterInfo.secretArn,
+    sql: tableQuery,
+    database: clusterInfo.dbName,
+  };
+
+  const tableQueryCommand = new ExecuteStatementCommand(tableQueryInput);
+  let tableQueryResponse;
+  try {
+    tableQueryResponse = await client.send(tableQueryCommand);
+  } catch (err) {
+    console.log(err);
+  }
+
+  const tables = tableQueryResponse?.records?.map((record) => record[0]?.stringValue) || [];
+
+  // Truncate each table
+  for (const table of tables) {
+    const truncateQuery = `TRUNCATE TABLE ${table} CASCADE;`;
```

Comment:
Add regex verifier to verify the table name prefix to prevent potential SQL injection. Now assertion before using table name in the SQL query template literal.